src/test/scala/sigmastate/utxo/ThresholdSpecification.scala

```diff
@@ -400,4 +401,21 @@ class ThresholdSpecification extends SigmaTestingCommons {
       twoToi *= 2
     }
   }
+
+  property("fail compilation when input limit exceeded") {
+    val proverA = new ErgoLikeTestProvingInterpreter
+    val skA = proverA.dlogSecrets.head
+    val pubkeyA = skA.publicImage
+    val keyName = "pubkeyA"
+    val env = Map(keyName -> pubkeyA)
+    val pubKeysStrExceeding = Array.fill[String](AtLeast.MaxChildrenCount + 1)(keyName).mkString(",")
+    an[CosterException] should be thrownBy compileWithCosting(env, s"""atLeast(2, Coll($pubKeysStrExceeding))""")
+    an[CosterException] should be thrownBy
+      compileWithCosting(env, s"""{ val arr = Coll($pubKeysStrExceeding); atLeast(2, arr) }""")
+
+    // max children should work fine
+    val pubKeysStrMax = Array.fill[String](AtLeast.MaxChildrenCount)(keyName).mkString(",")
+    compileWithCosting(env, s"""atLeast(2, Coll($pubKeysStrMax))""")
+    compileWithCosting(env, s"""{ val arr = Coll($pubKeysStrMax); atLeast(2, arr) }""")
+  }
```

Comment:
Please add test where `atLeast` uses some collection with unknown items, for example from `getVar`